flowmachine/flowmachine/features/subscriber/subscriber_event_count.py

```diff
@@ -0,0 +1,136 @@
+    direction : {'in', 'out', 'both'}, default 'out'
+        Whether to consider calls made, received, or both. Defaults to 'out'.
+    tables : str or list of strings, default 'all'
+        Can be a string of a single table (with the schema)
+        or a list of these. The keyword all is to select all
+        subscriber tables
+    kwargs :
+        Passed to flowmachine.EventTableSubset
+
+    Examples
+    --------
+
+    >>> s = SubscriberEventCount("2016-01-01", "2016-01-07", direction="in")
+    >>> s.get_dataframe()
+
+             subscriber  event_count
+    0  2ZdMowMXoyMByY07           65
+    1  MobnrVMDK24wPRzB           81
+    2  0Ze1l70j0LNgyY4w           57
+    3  Nnlqka1oevEMvVrm           63
+    4  4dqenN2oQZExwEK2           59
+                    ...          ...
+    """
+
+    def __init__(
+        self,
+        start,
+        stop,
+        subscriber_identifier="msisdn",
+        direction="both",
+        hours="all",
+        subscriber_subset=None,
+        tables="all",
+        **kwargs,
+    ):
+        self.start = start
+        self.stop = stop
+        self.subscriber_identifier = subscriber_identifier
+        self.direction = direction
+        if direction not in {"in", "out", "both"}:
+            raise ValueError("{} is not a valid direction.".format(self.direction))
+
+        try:
+            self.hours = kwargs["hours"]
+        except KeyError:
+            self.hours = "all"
+
+        if self.direction == "both":
+            column_list = [self.subscriber_identifier]
+            self.tables = tables
+        else:
+
+            column_list = [self.subscriber_identifier, "outgoing"]
+
+            if isinstance(tables, str) and tables.lower() == "all":
+                tables = [f"events.{t}" for t in self.connection.subscriber_tables]
+            elif type(tables) is str:
+                tables = [tables]
+            else:
+                tables = tables
+
+            self.tables = []
+            raise_warning = False
+            for t in tables:
+                if self._has_outgoing(t):
+                    self.tables.append(t)
+                else:
+                    raise_warning = True
+
+            if raise_warning:
+                warnings.warn(
+                    f"""Not all events table have a direction. Since you
+                    requested a directed count, only {self.tables} will be
+                    considered during the count.""",
+                    stacklevel=2,
+                )
+
+        self.unioned_query = EventsTablesUnion(
+            self.start,
+            self.stop,
+            tables=self.tables,
+            columns=column_list,
+            hours=hours,
+            subscriber_identifier=subscriber_identifier,
+            subscriber_subset=subscriber_subset,
+            **kwargs,
+        )
+        super().__init__()
+
+    def _has_outgoing(self, table):
+
+        return "outgoing" in Table(table).column_names
```

Comment:
Hmm, I'm not strictly opposed to having this in a helper method but it's quite succinct so might actually be more readable if it is inlined in L96 above (esp. since it doesn't actually use `self`, which confused me a little at first when I looked up the definition).